server/Dockerfile

```diff
@@ -0,0 +1,23 @@
+# Build the server files in a build stage
+FROM openjdk:8-jdk-alpine AS build
+# Copy the project into /tmp/server
+COPY . /tmp/server
+WORKDIR /tmp/server
+# Build the server with Gradle
+RUN ./gradlew assemble --no-daemon
+
+# Build the production server container
+FROM openjdk:8-jre-alpine
+EXPOSE 4567
+# Copy the distribution tar that was built in the build stage
+COPY --from=build /tmp/server/build/distributions/server.tar /app/
+WORKDIR /app
+# Untar the server files
+RUN tar -xvf server.tar
+
+## Add the wait script to the image
+ADD https://github.com/ufoscout/docker-compose-wait/releases/download/2.7.3/wait /wait
+RUN chmod +x /wait
```

Comment:
What is the wait script and why are we including it?

Reply:
Good idea. The wait command is so the server can wait for mongo to actually be ready before starting. If it starts before mongo it spews some errors (though it still works after mongo comes up).

Reply:
It is explained in `docker-compose.yml` with the `WAIT_HOSTS` key.

```yml
server:
    build: ./server # Location of the Dockerfile for the client
    restart: always
    environment:
      MONGO_ADDR: mongo # hostname of the mongo container
      MONGO_DB: prod
      WAIT_HOSTS: mongo:27017 # wait for mongo to start up before starting the server
    depends_on:
      - mongo
```

Reply:
It would still be nice to have a pointer here to some info on what the `wait` script does and why we need it. Having a comment in a totally different file isn't immediately helpful.